src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetPackageSearchService.cs

```diff
@@ -297,6 +383,17 @@ public void Dispose()
             return new MultiSourcePackageMetadataProvider(sourceRepositories, localRepo, globalRepo, new VisualStudioActivityLogger());
         }
 
+        private async ValueTask<IPackageMetadataProvider> GetPackageMetadataProviderAsync(
+            IReadOnlyCollection<PackageSourceContextInfo> packageSources,
+            IReadOnlyCollection<IProjectContextInfo> projects,
+            CancellationToken cancellationToken)
+        {
+            IReadOnlyCollection<SourceRepository> sourceRepositories = await _sharedServiceState.GetRepositoriesAsync(packageSources, cancellationToken);
+            SourceRepository localRepo = await _packagesFolderLocalRepositoryLazy.GetValueAsync(cancellationToken);
+            IEnumerable<SourceRepository> globalRepo = await GetAllPackageFoldersAsync(projects, cancellationToken);
```

Comment:
From comment above, you can add a condition to use globalRepo depending on packageSources argument.

```suggestion
            IEnumerable<SourceRepository> globalRepo = null;
            if (projects == null)
            {
                globalRepo = await GetAllPackageFoldersAsync(projects, cancellationToken);
            }
            else
            {
                globalRepo = = await _globalPackageFolderRepositoriesLazy.GetValueAsync(cancellationToken);
            }
```